mewarpx/tests/test_emission.py

```diff
@@ -672,3 +670,69 @@ def test_plasma_injector_fixedT2():
             df[label_base + key + '_std'] = np.std(val)
 
     assert testing_util.test_df_vs_ref(testname=name, df=df, margin=0.3)
+
+
+def test_arbitrary_distribution_emitter():
+    name = "arbitrary_distribution_emitter"
+    # Include a random run number to allow parallel runs to not collide.  Using
+    # python randint prevents collisions due to numpy rseed below
+    testing_util.initialize_testingdir(name)
+
+    # Initialize each run with consistent, randomly-chosen, rseed. Use a random
+    # seed instead for initial dataframe generation.
+    # np.random.seed(11874889)
+
+    run = diode_setup.DiodeRun_V1(
+        GEOM_STR='XZ', PERIOD=0.8e-06 * 512, D_CA=0.8e-06 * 512, DT=1e-12,
+        TOTAL_TIMESTEPS=1, DIAG_STEPS=1
+    )
+
+    run.setup_run(init_inert_gas=True, init_scraper=False, init_injectors=False)
+
+    z = np.arange(0, 512) + 0.5
+    sin_func = np.pi/2 * 1e13 * np.sin(np.pi / 512 * z)
+    plasma_density = np.tile(sin_func, (512, 1))
+
+    emitter = emission.ArbitraryDistributionVolumeEmitter(
+        spec_1_dgrid=plasma_density,
+        spec_2_dgrid=plasma_density,
+        T=1550, zmin=0, zmax=run.D_CA,
+    )
+
+    # Volume emission with spec'd plasma_density
+    emission.PlasmaInjector(
+        emitter=emitter, species1=run.electrons, species2=run.ions,
+        npart=512*512*10, plasma_density=1e19,
+    )
+
+    diags.FieldDiagnostic(
+        diag_steps=1, style='roelof', save_pdf=False,
+        plot=True,
+    )
+
+    diags.TextDiag(
+        diag_steps=1, preset_string='perfdebug'
+    )
+
+    run.init_warpx()
+    mwxrun.simulation.step(2)
+
+    electron_density = np.load(os.path.join(
+        os.curdir, "diags", "fields",
+        "electrons_particle_density_0000000002.npy"
+    ))
+
+    electron_xavg = np.mean(electron_density, axis=0)
+
+    ion_density = np.load(os.path.join(
```

Comment:
I'll discuss this in person, but this run doesn't check distinct e- and ion densities. I think that's fine with how I envision implementing it, but be aware you aren't testing it either way here.